experiment/simulation/js/integrate.js

```diff
@@ -0,0 +1,329 @@
+        if (k.classList.contains("high")) {
+            k.classList.remove("high")
+            k.classList.add("low")
+            k.innerHTML = 'Input 1<br>0'
+            listInput[0].input = 0;
+        } else {
+            k.classList.remove("low")
+            k.classList.add("high")
+            k.innerHTML = 'Input 1<br>1'
+            listInput[0].input = 1;
+        }
+    })
+    d.input = 0;
+    d.voltage = 5;
+    const container = document.getElementById("diagram");
+
+    const d1 = new Object();
+    d1.id = id;
+    d1.input = 1;
+    d1.voltage = 5;
+
+    container.insertAdjacentElement("afterbegin", d);
+
+    jsplumbInstance.draggable(id, { "containment": true });
+    listInput.push(d1);
+
+    addInstanceFinalInput(id);
+
+}
+
+function compInput1() {
+    const id = "input1";
+    const d = document.createElement('div');
+    d.innerHTML = 'Input 2<br>1'
+    d.id = id;
+    d.className = 'io-component';
+    d.style.top = "9rem";
+    d.style.left = "0.625rem";
+    d.classList.add("high");
+    d.addEventListener("dblclick", () => {
+        k = document.getElementById("input1")
+        if (k.classList.contains("high")) {
+            k.classList.remove("high")
+            k.classList.add("low")
+            k.innerHTML = 'Input 2<br>0'
+            listInput[1].input = 0;
+        } else {
+            k.classList.remove("low")
+            k.classList.add("high")
+            k.innerHTML = 'Input 2<br>1'
+            listInput[1].input = 1;
+        }
+    })
+    d.addEventListener("long-press", () => {
+        k = document.getElementById("input1")
+        if (k.classList.contains("high")) {
+            k.classList.remove("high")
+            k.classList.add("low")
+            k.innerHTML = 'Input 2<br>0'
+            listInput[1].input = 0;
+        } else {
+            k.classList.remove("low")
+            k.classList.add("high")
+            k.innerHTML = 'Input 2<br>1'
+            listInput[1].input = 1;
+        }
+    })
+    d.input = 0;
+    d.voltage = 5;
+    const container = document.getElementById("diagram");
+
+    const d1 = new Object();
+    d1.id = id;
+    d1.input = 1;
+    d1.voltage = 5;
+
+    container.insertAdjacentElement("afterbegin", d);
+
+    jsplumbInstance.draggable(id, { "containment": true });
+    listInput.push(d1);
+
+    addInstanceFinalInput(id);
+
+}
+
+function compOutput() {
+    const id = "output0";
+    const d = document.createElement('div');
+    d.innerHTML = 'Output<br>-'
+    d.id = id;
+    d.className = 'io-component';
+    d.style.top = "1.25rem";
+    d.style.right = "0.625rem";
+    d.outputsign = 1;
+    d.voltage = 0;
+
+    const d1 = new Object();
+    d1.id = id;
+    d1.voltage = 0;
+    d1.outputsign = 1;
+    const container = document.getElementById("diagram");
+
+
+    container.insertAdjacentElement("afterbegin", d);
+
+    jsplumbInstance.draggable(id, { "containment": true });
+    listOutput.push(d1);
+
+    addInstanceFinalOutput(id);
```

Comment:
This file, has almost no comments and the variable names in a few places are almost making it impossible to understand what the code is doing. 
Please add some comments, and use better naming

Reply:
Wrote some comments already in exp - 1. It's the same here.